### PR TITLE
refactor docker_cli_netmode_test .go with Assert statement

### PR DESCRIPTION
**Pull request by cxxly**
part of #16756 

Signed-off-by: Xiaoxu Chen chenxiaoxu14@otcaix.iscas.ac.cn

### CODE REVIEW

File: integration-cli/docker_cli_netmode_test.go [hunk 1]
```diff
@@ -32,52 +24,45 @@ func (s *DockerSuite) TestNetHostname(c *check.C) {
 	)
 
 	runCmd = exec.Command(dockerBinary, "run", "-h=name", "busybox", "ps")
-	if out, _, err = runCommandWithOutput(runCmd); err != nil {
-		c.Fatalf(out, err)
-	}
-	checkContains(stringCheckPS, out, c)
+	out, _, err = runCommandWithOutput(runCmd)
```
Comment (HuKeping) [inline integration-cli/docker_cli_netmode_test.go]: How about using `dockerCmd`, it will check the `err`

Comment (vdemeester) [inline integration-cli/docker_cli_netmode_test.go]: Yep, this could use `dockerCmd` :wink:

File: integration-cli/docker_cli_netmode_test.go [hunk 2]
```diff
@@ -32,52 +24,45 @@ func (s *DockerSuite) TestNetHostname(c *check.C) {
...
+	c.Assert(err, checker.IsNil, check.Commentf("%v", out))
+	c.Assert(out, checker.Contains, stringCheckPS)
 
 	runCmd = exec.Command(dockerBinary, "run", "--net=host", "busybox", "ps")
-	if out, _, err = runCommandWithOutput(runCmd); err != nil {
-		c.Fatalf(out, err)
-	}
-	checkContains(stringCheckPS, out, c)
+	out, _, err = runCommandWithOutput(runCmd)
```
Comment (vdemeester) [inline integration-cli/docker_cli_netmode_test.go]: Could use `dockerCmd` here too

File: integration-cli/docker_cli_netmode_test.go [hunk 3]
```diff
@@ -32,52 +24,45 @@ func (s *DockerSuite) TestNetHostname(c *check.C) {
...
+	c.Assert(err, checker.IsNil, check.Commentf("%v", out))
+	c.Assert(out, checker.Contains, stringCheckPS)
 
 	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=bridge", "busybox", "ps")
-	if out, _, err = runCommandWithOutput(runCmd); err != nil {
-		c.Fatalf(out, err)
-	}
-	checkContains(stringCheckPS, out, c)
+	out, _, err = runCommandWithOutput(runCmd)
```
Comment (vdemeester) [inline integration-cli/docker_cli_netmode_test.go]: Could use `dockerCmd` here too

File: integration-cli/docker_cli_netmode_test.go [hunk 4]
```diff
@@ -32,52 +24,45 @@ func (s *DockerSuite) TestNetHostname(c *check.C) {
...
+	c.Assert(err, checker.IsNil, check.Commentf("%v", out))
+	c.Assert(out, checker.Contains, stringCheckPS)
 
 	runCmd = exec.Command(dockerBinary, "run", "-h=name", "--net=none", "busybox", "ps")
-	if out, _, err = runCommandWithOutput(runCmd); err != nil {
-		c.Fatalf(out, err)
-	}
-	checkContains(stringCheckPS, out, c)
+	out, _, err = runCommandWithOutput(runCmd)
```
Comment (vdemeester) [inline integration-cli/docker_cli_netmode_test.go]: Could use `dockerCmd` here too

**Comment by cxxly** [comment]
@vdemeester @HuKeping  great, I will update

**Comment by cxxly** [comment]
At this point, I add a function dockerCmdWithFail which call dockerCmdWithError, maybe we should refactor dockerCmdWithError with err checking.

**Comment by cxxly** [comment]
@vdemeester @HuKeping  Are there any other ideas

**Comment by vdemeester** [comment]
Let's not refactor too much for now `dockerCmdWithError` might not be used with `checker.NotNil` each time.

LGTM 🐰

File: integration-cli/docker_cli_netmode_test.go [hunk 5]
```diff
@@ -14,152 +12,80 @@ import (
...
-// checkContains is a helper function that validates a command output did
-// contain what was expected.
-func checkContains(expected string, out string, c *check.C) {
-	if !strings.Contains(out, expected) {
-		c.Fatalf("Expected '%s', got '%s'", expected, out)
-	}
+// DockerCmdWithFail executes a docker command that is supposed to fail and returns
+// the output, the exit code. If the command returns an Nil error, it will fail and
+// stop the tests.
+func dockerCmdWithFail(c *check.C, args ...string) (string, int) {
```
Comment (runcom) [inline integration-cli/docker_cli_netmode_test.go]: why not exposing this in docker_utils? This pattern is often used elsewhere also

Comment (vdemeester) [inline integration-cli/docker_cli_netmode_test.go]: @runcom maybe but I think this is not the right PR to do it. I'm gonna work on this (with starting integration test with go & co) but I think we shouldn't have not too much `dockerCmdWith..` function :stuck_out_tongue:

**Comment by runcom** [comment]
LGTM

**Pull request merged**